### ISSUE TITLE
Backup rates shipping origin and per store config bugs

### ISSUE THREAD

**Issue by andreas-ateles** [taxjar/taxjar-magento-extension#2]
It seems that most (or all) of the code that has to do with backup rates configuration and sync always uses the global default shipping origin setting (Current Configuration Scope: Default Config), instead of the per store configured value. This prevents the use of backup rates feature for international Magento stores that are based outside of USA but have a store in USA.

(The backup rates feature was one of the reasons for choosing TaxJar over the alternatives, so I hope this can be fixed.)

Example:

International Magento store based in Sweden. Default shipping origin in Sweden. One website/store per country. Default store has TaxJar disabled among with backup rates.

The USA store has a shipping origin set properly in the USA. TaxJar enabled and backup rates enabled for it. But the button for "Sync Backup Rates" does not show here, and no backup rates are ever synchronized to Magento.

In the extension code I could see the following reason for this: The code that checks the origin address is executed in the admin settings scope (store id 0, default), and not at the USA store level which is currently selected in the settings configuration scope. The same thing happens when the backup rates sync cronjob i executed, it is running in the store id 0 scope (and only for that store).

(As you have made the backup rates setting available per store, with possible different shipping origin addresses, I'm not sure how you have thought that backup rates would work in such cases.)

**Comment by fastdivision** [comment]
Hey @andreas-ateles! Thanks for the detailed example. We've had several international merchants request this improvement and it's something I'd like to address soon. Ideally, I'd like to drop the shipping origin requirement altogether for pulling backup rates from our API. We use it for new merchants that may not have any existing nexus addresses in TaxJar and to set default tax configuration values, but it doesn't seem necessary. I'd rather just use it for checkout calculations where the shipping origin is already based on the current store. Seems like a much cleaner approach than having to manage scope data for the cron job.

Let's keep this issue open and I'll start a PR once we make these tweaks to our API endpoint.

**Comment by andreas-ateles** [comment]
So shipping origin is just used as a nexus in this case? In that case it does indeed seem more reasonable to just use the nexus configuration. Sounds great!

Any idea on possible timeline for this change? (To help us decide what to do in the meantime.)

**Comment by fastdivision** [comment]
@andreas-ateles Yes the shipping origin is used as a default nexus for backup rates in case a new merchant doesn't have any nexus addresses set up in TaxJar. Now that we can manage nexus addresses directly in Magento, I don't see any reason why we can't tell the merchant to configure their nexus addresses prior to pulling down rates.

I'm working on a new backup rates API endpoint with shipping origin params optional and support for custom sourcing logic (origin vs. destination). I'd say optimistically within a month after some additional testing and QA, since this requires an update to both our API and extension.
